.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,50 @@
+name: Build with emscripten
+
+on: push
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install cached emscripten
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: 'emsdk'
+          key: 'emsdk'
+
+      - name: Install emscripten
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          curl -L https://github.com/emscripten-core/emsdk/archive/HEAD.tar.gz | tar xz
+          mv emsdk-* emsdk
+          ./emsdk/emsdk install latest
+          ./emsdk/emsdk activate latest
+
+      - name: Build blink
+        run: |
+          . ./emsdk/emsdk_env.sh
+          emmake make -j$(nproc)
+          curl -sSL https://raw.githubusercontent.com/gzuidhof/coi-serviceworker/master/coi-serviceworker.min.js -o o//blink/blink-web/coi-serviceworker.min.js --create-dirs
+          emcc -O2 o//blink/blink.o o//blink/blink.a -lm -pthread -lrt -o o//blink/blink-web/blink.html \
+            --shell-file blink/blink-shell.html \
+            -s INVOKE_RUN=0 \
+            -s INITIAL_MEMORY=1073741824 \
+            -s EXIT_RUNTIME=1 \
+            -s EXPORTED_FUNCTIONS=_main \
+            -s EXPORTED_RUNTIME_METHODS=setValue,getValue,stringToUTF8Array,lengthBytesUTF8,cwrap \
+            -fno-builtin-exit \
+            -sASSERTIONS
+
+      - name: Deploy sample to GitHub pages
+        uses: JamesIves/github-pages-deploy-action@v4
+        # if: ${{ github.ref == 'refs/heads/master' }}
```

Comment:
Please pin a version here too if possible.

Reply:
You mean pinning a specific commit?

Reply:
Done. Pinned to v4.4.1